functions/utils/getCollection.ts

```diff
@@ -0,0 +1,21 @@
+import { CollectionDocument } from '../../src/graphql/data/__generated__/types-and-hooks'
+import client from '../client'
+
+export default async function getCollection(collectionAddress: string, url: string) {
+  const { data } = await client.query({
+    query: CollectionDocument,
+    variables: {
+      addresses: collectionAddress,
+    },
+  })
+  const collection = data?.nftCollections?.edges[0]?.node
+  if (!collection || !collection.name) {
+    return undefined
+  }
```

Comment:
this behaves differently than the asset one right? should we return undefined in asset if we can't find a name / info ?

Reply:
From what I've looked into, a lot of assets don't actually have names which is why I had to create the weird branching naming structure. For collections we can do this since for some reason the GraphQL indexes every single valid address as an `NFTContract` even if that is not the case, so the only way we can differentiate whether or not a collection is in fact an `NFTContract` is through the name.